src/components/Input/TextArea.jsx

```diff
@@ -43,17 +43,11 @@ export const TextArea = ({
     }
     return cleanedSplit.length || 0
   }, [value])
-  useEffect(() => {
-    // debounced with setTimeout
-    const handler = setTimeout(() => {
-      setIsLengthExceeded(
-        maxLength ? value && value.length > maxLength : maxWords ? getWords() > maxWords : false
-      )
-    }, 500)
 
-    return () => {
-      clearTimeout(handler)
-    }
+  useEffect(() => {
+    setIsLengthExceeded(
+      maxLength ? value && value.length > maxLength : maxWords ? getWords() > maxWords : false
+    )
```

Comment:
got rid of the debounce because we're conditionally showing error messages. the debounce caused a delay -> resulting in the `isLengthExceeded` error message to flash when we go from over word count to under word count